File: array/valid_mountain_array.py
# Given an array of integers arr, return true if and only if it is a valid mountain array.

# Recall that arr is a mountain array if and only if:

# arr.length >= 3
# There exists some i with 0 < i < arr.length - 1 such that:
# arr[0] < arr[1] < ... < arr[i - 1] < arr[i]
# arr[i] > arr[i + 1] > ... > arr[arr.length - 1]

def valid_mountain(arr):
    i, j, n = 0, len(arr)-1, len(arr)
    while i+1 < n and arr[i+1] > arr[i]:
        i += 1
    while j > 0 and arr[j] < arr[j-1]:
        j -= 1
    return 0 < i == j < n-1


if __name__ == '__main__':
    arr = [1, 2, 3, 2]
    print(valid_mountain(arr))

File: array/validMountainArray.js
// Given an array of integers arr, return true if and only if it is a valid mountain array.

// Recall that arr is a mountain array if and only if:
// arr.length >= 3
// There exists some i with 0 < i < arr.length - 1 such that:
// arr[0] < arr[1] < ... < arr[i - 1] < arr[i]
// arr[i] > arr[i + 1] > ... > arr[arr.length - 1]

// Input: arr = [2,1]
// Output: false

// Input: arr = [2,1]
// Output: false

// Input: arr = [3,5,5]
// Output: false

function isMountain(arr) {
    if (arr.length < 3) return false
    let i = 0, j = arr.length - 1
    while (i < arr.length && arr[i + 1] > arr[i]) {
        i++
    }
    console.log(i)
    while (j > i && arr[j - 1] > arr[j]) {
        j--
    }
    console.log(j)
    return i === j && i != arr.length - 1 && j != 0
}

let arr = [1, 2, 3, 2]
console.log(isMountain(arr))